backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java

```diff
@@ -19,11 +21,64 @@ public UserService(UserDb userDb) {
         this.userDb = userDb;
     }
 
-    public Optional<List<Game>> listAllGames(String id) {
+    public List<Game> listAllGames(String id) {
         Optional<User> user = userDb.findById(id);
         if (user.isPresent()) {
-            return Optional.of(user.get().getPlayedGames());
+            return user.get().getPlayedGames();
         }
-        return Optional.empty();
+        return List.of();
     }
+
+    public List<GameSession> listAllSessions(String id) {
+        Optional<User> user = userDb.findById(id);
+        if (user.isPresent()) {
+            List<Game> gameList = user.get().getPlayedGames();
+            return gameList.stream().flatMap(game -> game.getGameSessionList().stream())
+                    .collect(Collectors.toList());
+        }
+        return List.of();
```

Comment:
you may want to consider throwing an exception here if a user does not exist 
`throw new ResponseStatusException()`